checkov/common/graph/db_connectors/igraph/igraph_db_connector.py

```diff
@@ -0,0 +1,73 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, TypeVar
+
+from igraph import Graph
+
+from checkov.common.graph.db_connectors.db_connector import DBConnector
+from checkov.common.graph.graph_builder import CustomAttributes
+
+if TYPE_CHECKING:
+    from checkov.common.graph.graph_builder.graph_components.blocks import Block  # noqa
+    from checkov.common.graph.graph_builder.local_graph import LocalGraph
+
+_Block = TypeVar("_Block", bound="Block")
+
+
+class IgraphConnector(DBConnector[Graph]):
+    def __init__(self) -> None:
+        self.graph = Graph(directed=True)
+
+    def save_graph(self, local_graph: LocalGraph[_Block], add_bulk_edges: bool = False) -> Graph:
+        return self.networkit_from_local_graph(local_graph)
+
+    def get_reader_endpoint(self) -> Graph:
+        return self.graph
+
+    def get_writer_endpoint(self) -> Graph:
+        return self.graph
+
+    def networkit_from_local_graph(self, local_graph: LocalGraph[_Block]) -> Graph:
+        # colors = {
+        #     "Pod": "blue",
+        #     "Deployment": "magenta",
+        #     "Service": "olive",
+        # }
+        #
+        # random_colors = ("red", "blue", "olive", "gold", "magenta", "purple", "black")
+        # random_colors_len = len(random_colors)
+        # randrange(len(random_colors))
```

Comment:
is this relevant?

Reply:
Can be used for visualization, so I'll leave it here for now

Reply:
that was my left over, which I wanted to use for adding kind of debugging capabilities